rotkehlchen/api/rest.py

```diff
@@ -5583,3 +5587,69 @@ def fetch_token_balance_for_address(
             return wrap_in_fail_result(str(e), status_code=HTTPStatus.CONFLICT)
 
         return _wrap_in_ok_result(result=balance)
+
+    @async_api_call()
+    def migrate_solana_token(
+            self,
+            old_asset: 'CryptoAsset',
+            address: 'SolanaAddress',
+            decimals: int,
+            token_kind: 'SOLANA_TOKEN_KINDS',
+    ) -> dict[str, Any]:
+        """This is a temporary endpoint to correct custom user input
+        solana tokens input before release 1.40.
+
+        Creates a new solana token with the provided address and metadata,
+        replaces all references in the database and cleans up the migration table if necessary.
+        """
+        with GlobalDBHandler().conn.read_ctx() as cursor:
+            if cursor.execute(
+                'SELECT COUNT(*) FROM user_added_solana_tokens WHERE identifier = ?',
+                (old_asset.identifier,),
+            ).fetchone()[0] == 0:
+                return wrap_in_fail_result(
+                    message='Token does not exist in user_added_solana_tokens table',
+                    status_code=HTTPStatus.CONFLICT,
+                )
+        with self.migration_lock:
+            try:
+                GlobalDBHandler.add_asset(solana_token := SolanaToken.initialize(
+                    address=address,
+                    token_kind=token_kind,
+                    decimals=decimals,
+                    name=old_asset.name,
+                    symbol=old_asset.symbol,
+                    started=old_asset.started,
+                    forked=old_asset.forked,
+                    swapped_for=old_asset.swapped_for,
+                    coingecko=old_asset.coingecko,
+                    cryptocompare=old_asset.cryptocompare,
+                ))
+            except InputError as e:
+                return wrap_in_fail_result(str(e), status_code=HTTPStatus.CONFLICT)
+
+            try:
+                with self.rotkehlchen.data.db.conn.write_ctx() as write_cursor:
+                    self.rotkehlchen.data.db.add_asset_identifiers(
+                        write_cursor=write_cursor,
+                        asset_identifiers=[solana_token.identifier],
+                    )
+                self.rotkehlchen.data.db.replace_asset_identifier(
```

Comment:
What if this raises? Won't you also need to then delete the asset from the user db?